### PR TITLE
Basic database setup

### PR DESCRIPTION
**Pull request by mgerhold**

### CODE REVIEW

**Comment by tinsever** [comment]
lgtm!

**Review by peddermaster2 — APPROVED** [review]

File: app/Controllers/User.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    public function show(int $id)
+    {
+        $model = model(UserModel::class);
+
+        $data = $model->getUser($id);
+
+        if ($data) {
+            return $this->response->setJSON($data);
+        } else {
```
Comment (PurpurTentakel97) [inline app/Controllers/User.php:23]: unnötiges else

File: app/Database/Migrations/2024-08-10-143100_AddUser.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                'null' => false,
+            ],
+            'time_updated' => [
+                'type' => 'DATETIME',
+                'null' => false,
+                'default' => new RawSql('CURRENT_TIMESTAMP'),
+                'on_update' => new RawSql('CURRENT_TIMESTAMP'),
+            ],
+        ]);
+        $this->forge->addKey('id', true);
```
Comment (PurpurTentakel97) [inline app/Database/Migrations/2024-08-10-143100_AddUser.php:34]: consider using `addPrimaryKey()` instead to make clear its a primary key.

**Review by PurpurTentakel97 — CHANGES_REQUESTED** [review]

File: app/Models/UserModel.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    /**
+     * @param $id bool|int $id
+     * @return array|object|null
+     */
+    public function getUser(bool|int $id = false): object|array|null
+    {
+        if ($id === false) {
+            return $this->findAll();
+        } else {
```
Comment (PurpurTentakel97) [inline app/Models/UserModel.php:19]: unnecessary else

Comment (mgerhold) [reply app/Models/UserModel.php]: changed

File: app/Database/Migrations/2024-08-10-143100_AddUser.php (hunk 2 above)
Comment (mgerhold) [reply app/Database/Migrations/2024-08-10-143100_AddUser.php]: Implemented the change, thanks! :)

File: app/Controllers/User.php (hunk 1 above)
Comment (mgerhold) [reply app/Controllers/User.php]: changed

**Review by PurpurTentakel97 — APPROVED** [review]

**Pull request merged**